app/src/services/webtoon/episodeService.js

```diff
@@ -14,15 +14,44 @@ class EpisodeService {
     return episodes;
   }
 
-  async getEpisodeDetail(episodeId, userId) {
-    const episode = await this.episodeRepository.getEpisodeDetailById(episodeId, userId);
-    if (!episode) {
-      throw new CustomError("해당 에피소드를 찾을 수 없습니다.", 404);
-    }
-    if (episode.hasRated !== undefined && episode.hasRated !== null) {
-      episode.hasRated = !!episode.hasRated;
+  async openEpisodeDetail(episodeId, userId) {
+    let connection;
+    try {
+      connection = await pool.getConnection();
+
+      await connection.beginTransaction();
+
+      const episode = await this.episodeRepository.getEpisodeDetailById(
+        connection,
+        episodeId,
+        userId
+      );
+
+      if (!episode) {
+        await connection.rollback();
+        throw new CustomError("해당 에피소드를 찾을 수 없습니다.", 404);
+      }
+      const webtoonId = episode.webtoonId;
+      if (!webtoonId) {
+        await connection.rollback();
+        throw new CustomError("에피소드에 연결된 웹툰 정보를 찾을 수 없습니다.", 500);
+      }
```

Comment:
1. 해당 코드블럭에서는 connection을 통해 write 동작을 수행하지 않았기때문에 rollback이 필요하지 않아요
2. try 블록에서 에러가 throw 됐기때문에 어처피 catch 블록으로 가서 rollback 코드가 실행됩니다

```suggestion
      if (!episode) {
        throw new CustomError("해당 에피소드를 찾을 수 없습니다.", 404);
      }
      const webtoonId = episode.webtoonId;
      if (!webtoonId) {
        throw new CustomError("에피소드에 연결된 웹툰 정보를 찾을 수 없습니다.", 500);
      }
```

Reply:
피드백 감사합니다! 말씀해주신 대로 if문에서는 write 동작이 없어서 rollback이 필요 없네요. 어차피 throw 되면 catch에서 rollback이 처리되니, 중복으로 호출하고 있었네요.